libs/ledger-live-common/src/hw/getTransport.ts

```diff
@@ -0,0 +1,60 @@
+import { Subject } from "rxjs";
+import { withDevice } from "./deviceAccess";
+
+export type BidirectionalEvent =
+  | {
+      type: "input-frame";
+      apduHex: string;
+    }
+  | {
+      type: "device-request";
+      data: string;
+    }
+  | {
+      type: "device-response";
+      data: string;
+    }
+  | {
+      type: "error";
+      error: unknown;
+    };
+
+export type GetTransportRequest = {
+  deviceId: string;
+};
+
+export default function getTransport({
+  deviceId,
+}: GetTransportRequest): Subject<BidirectionalEvent> {
+  const subject = new Subject<BidirectionalEvent>();
+
+  withDevice(deviceId)((transport) => {
+    // The input part of the bidirectional communication
+    subject.subscribe({
+      next: (e) => {
+        /**
+         * Receiving an event from the ipc bridge allows us to pass a msg
+         * into an ongoing withDevice job. Allowing to exchange messages with the
+         * transport exposed from the job.
+         */
+        if (e?.type === "input-frame") {
+          subject.next({ type: "device-request", data: e.apduHex });
+          // TODO important avoiding collisions, also types will be broken
+          transport
+            .exchange(Buffer.from(e.apduHex, "hex"))
+            .then((response) =>
+              subject.next({
+                type: "device-response",
+                data: response.toString("hex"),
+              })
+            )
+            .catch((error) => subject.next({ type: "error", error }));
+        }
+      },
+    });
+
+    return subject;
+  }).toPromise();
```

Comment:
Why converting the `job` function to a promise?
The `job` function returns an `Observable`, not a promise 🤔

```ts
const withDevice: (deviceId: string) => <T>(job: (t: Transport) => Observable<T>) => Observable<T>
```

cf. https://github.com/LedgerHQ/ledger-live/blob/e58610f63dfa3ae1d213741a063390971a7e1587/libs/ledger-live-common/src/hw/deviceAccess.ts#L83

Since subject is an `Observable`, shouldn't it be like that instead:

```ts
  withDevice(deviceId)((transport) => {
    // The input part of the bidirectional communication
    subject.subscribe({
      ...
    })

    return subject;
  });
```

Reply:
> Why converting the job function to a promise?

The call to `toPromise()` subscribes to the observable and calls the inner function, if we don't do that nothing happens.

The subject is then returned because it is actually the observable you receive from `withDevice` but we need the Subject interface.

Reply:
ideally , this should return an instance of `Transport` because what you call `Subject<BidirectionalEvent>` sounds equivalent to a Transport. (which also have a "disconnect" event system for errors, etc..)

saying this here because i'm actually going to need this soon for https://ledgerhq.atlassian.net/wiki/spaces/WALLETCO/pages/3948118370/tech+debt+documentation

Reply:
actually i see you need it for the "command" serialisation, i think i would just reword this to something else to avoid confusion.

maybe "openTransportAsSubject" or something.

ideally we'll then be able to "create a transport from a command" on the renderer side 🤔

Reply:
Understood, quick question @gre, is it fine for you if we do the changes incrementally after having this merged ?

Reply:
i think it's ok, but it could be interesting to confirm together this model will be future proof and compatible with the idea to rebuild a `Transport` instance on the renderer side :)